pkg/datasources/account_roles_acceptance_test.go

```diff
@@ -0,0 +1,86 @@
+package datasources_test
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAcc_AccountRoles_basic(t *testing.T) {
+	accountRoleNamePrefix := "account_roles_test_prefix_"
+	accountRoleName1 := accountRoleNamePrefix + strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	accountRoleName2 := accountRoleNamePrefix + strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	accountRoleName3 := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	comment := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	configVariables := config.Variables{
+		"account_role_name_1": config.StringVariable(accountRoleName1),
+		"account_role_name_2": config.StringVariable(accountRoleName2),
+		"account_role_name_3": config.StringVariable(accountRoleName3),
+		"pattern":             config.StringVariable(accountRoleNamePrefix + "%"),
+		"comment":             config.StringVariable(comment),
+	}
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: config.TestNameDirectory(),
+				ConfigVariables: configVariables,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.snowflake_account_roles.test", "roles.#", "2"),
+					containsAccountRole(accountRoleName1, comment),
+					containsAccountRole(accountRoleName2, comment),
+					func(state *terraform.State) error {
+						err := containsAccountRole(accountRoleName3, comment)(state)
```

Comment:
isnt this just !containsAccountRole? i guess you still have error message to deal with. maybe just putting it into a function called stateNotContainsAccountRole? or adding a boolean flag to containsAccountRole for whether or not the role is expected to be present

Reply:
done